birdhouse/components/stac/config/magpie/stac_hooks.py

```diff
@@ -81,64 +64,84 @@ def create_item_resource(response):
-                                        item_res_create = False
-                                        break
-
-        # create resource /stac/stac/collections/<collection_id>/items if does not already exist
-        if item_res_create:
-            item_res = ru.create_resource("items", None, Route.resource_type_name, collection_res_id, db_session=session)
-            item_res_id = item_res.json["resource"]["resource_id"]
-
-        # create resource stac/stac/collection/<collection_id>/items/<item_id>
-        # In try catch since Magpie resource are persistent but stac-db could be wiped
-        try:
-            item_res = ru.create_resource(item_id, display_name, Route.resource_type_name, item_res_id, db_session=session)
-        except Exception as exc:
-            LOGGER.warning("Failed creation of the item %s %s", display_name, str(exc), exc_info=exc)
-        session.commit()   
-   
+        # Create the resource tree
+        create_resource_tree(f"stac/collections/{collection_id}/items/{item_id}", children, 0, service.resource_id, session, display_name)
+        session.commit()
+
     except Exception as exc:
         LOGGER.error("Unexpected error while creating the item %s %s", display_name, str(exc), exc_info=exc)
-    
+        session.rollback()
+        return response
+
     return response
 
 def extract_display_name(links):
-    # type: (JSON array) -> string
+    # type: (List[Dict[str, str]]) -> str
     """
     Extract THREDD path from a STAC links
     """
     display_name = None
     for link in links:
-        if link['rel'] == 'source':
+        if link["rel"] == "source":
             # Example of title `thredds:birdhouse/CMIP6` -> `birdhouse/CMIP6`
-            display_name = link['title'].split(':')[1]
+            display_name = link["title"].split(":")[1]
             break
+    if display_name is None:
+        raise Exception("The display name was not extracted properly")
 
     return display_name 
+
+def create_resource_tree(resource_tree, nodes, current_depth, parent_id, session, display_name):
+    # type: (str, OrderedDict, int, int, session, str) -> None 
+    """
+    Create the resource tree on Magpie
+    """
+    tree = resource_tree.split("/")
+    # We are at the max depth of the tree.
+    if current_depth > len(tree) - 1:
+        return 
+
+    resource = tree[current_depth]
+
+    create_resource = True
+    for childs in nodes.values():
+        # Find the resource in the current children.
+        if childs["node"].resource_name == resource:
+            children = childs["children"]
+            # Since the resource exists, we can use its id to create the next resource.
+            parent_id = childs["node"].resource_id
+            next_depth = current_depth + 1
+            return create_resource_tree(resource_tree, children, next_depth, parent_id, session, display_name)
+            create_resource = False
+            break
```

Comment:
The two lines after `return` will never execute. 
The `return` on the same line as `created_resource_tree`, although valid, is misleading with nothing (`None`) returned.